include/VSSModel.hpp

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <cmath>
+#include <functional>
+#include <limits>
+#include <stdexcept>
+#include <vector>
+
+namespace cda_rail::vss {
+using SeparationFunction = std::function<double(size_t, size_t)>;
+
+enum class ModelType {
+  Discrete    = 0,
+  Continuous  = 1,
+  Inferred    = 2,
+  InferredAlt = 3
+};
+
+namespace functions {
+[[nodiscard]] static double uniform(size_t i, size_t n) {
+  double ret_val = (static_cast<double>(i) + 1) / static_cast<double>(n);
+  if (ret_val > 1) {
+    ret_val = 1;
+  }
+  return ret_val;
+}
+
+[[nodiscard]] static double chebyshev(size_t i, size_t n) {
+  if (i >= n - 1) {
+    return 1;
+  }
+
+  const auto   n_points = static_cast<double>(n) - 1;
+  const auto   k        = n_points - static_cast<double>(i);
+  const double pi       = 3.14159265358979323846;
```

Comment:
Just a comment because you might not be aware. These kinds of constants you can declare as `constexpr` which makes it a compile-time constant. Especially in a static method, this has negligible overhead so no need to change it really.

Reply:
done